bitcoin/src/blockdata/witness.rs

```diff
@@ -398,6 +398,40 @@ impl Witness {
             }
         })
     }
+
+    /// Get the raw taproot control block following BIP341 rules.
+    ///
+    /// This does not guarantee that this represents a P2TR [`Witness`]. It
+    /// merely gets the last or second to last element depending on the first
+    /// byte of the last element being equal to 0x50. See
+    /// [Script::is_p2tr](crate::blockdata::script::Script::is_p2tr) to
+    /// check whether this is actually a Taproot witness.
+    pub fn raw_taproot_control_block(&self) -> Option<&[u8]> {
+        self.last().and_then(|last| {
+            // From BIP341:
+            // If there are at least two witness elements, and the first byte of
+            // the last element is 0x50, this last element is called annex a
+            // and is removed from the witness stack.
+            if self.len() >= 3 && last.first() == Some(&TAPROOT_ANNEX_PREFIX) {
+                self.nth(self.len() - 2)
+            } else if self.len() >= 2 {
+                Some(last)
+            } else {
+                None
+            }
+        })
+    }
```

Comment:
I think we could extract this into `fn control_block_index` and then use that to index both control block and tapscript. (or `tapscript_index`, I don't care). Wouldn't do that with annex of course.

Reply:
I don't think this makes sense.

Reply:
Why not? It de-duplicates the logic which is the same. The number is just shifted by one.